### PR TITLE
Gi 64 test hardware controller

### PR DESCRIPTION
**Pull request by darkogoluza**

### CODE REVIEW

File: tests/src/hardware_controller_test.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    void TearDown() override {}
+};
+
+TEST_F(HardwareControllerTest, hc_handler_initialise)
+{
+    hc_handler_initialise(3, 3, 3);
+
+
```
Comment (wikcioo) [inline tests/src/hardware_controller_test.cpp:45]: Keep one line separation, not two

**Review by wikcioo — CHANGES_REQUESTED** [review]

File: tests/src/hardware_controller_test.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+TEST_F(HardwareControllerTest, hc_handler_task_run3)
+{
+    hc_handler_task_run(4);
+
+    ASSERT_EQ(xMessageBufferSend_fake.call_count, 0);
+}
+
+TEST_F(HardwareControllerTest, hc_handler_task_run4)
+{
+    hc_handler_task_run(6);
```
Comment (wikcioo) [inline tests/src/hardware_controller_test.cpp:137]: We never get to counter = 6, because we reset at 5. Perhaps unnecessary test?

**Review by wikcioo — CHANGES_REQUESTED** [review]

File: tests/src/hardware_controller_test.cpp (hunk 2 above)
Comment (darkogoluza) [reply tests/src/hardware_controller_test.cpp:137]: Well that is why is the test there to ensure that we never get to 6

Comment (wikcioo) [reply tests/src/hardware_controller_test.cpp:137]: How do you ensure that we never get to 6?

**Review by wikcioo — APPROVED** [review]

**Pull request merged**